syntax="proto3";

package GameMessage;


//登录（进入游戏）
message LoginRequest {
	 int32 msgId=1;
	 string account                 = 2; //账号
	 string password                = 3; //密码
     LoginType loginType            = 4; //登录类型 
	 int64 sessionId				= 5; //会话ID（服务器内部使用）
	 int32 gateId					= 6; //网关ID（服务器内部使用）
	 string ip						= 7; //IP地址（服务器内部使用）
	 string version					= 8; //版本号
}
message LoginResponse {
	int32 msgId=1;
	 bool isOk 			= 2; //请求是否发送成功
	 int64 uid 			= 3; //用户id
	 int64 rid			= 4; //角色ID
	 int64 sessionId	= 5; //会话ID
}

//退出游戏
message QuitRequest{
	int32 msgId=1;
	 int64 rid 			= 2; //角色ID
}

//退出游戏
message QuitResponse{
	int32 msgId=1;
	int32 result		= 2; //结果
}



//登录子游戏
message LoginSubGameRequest{
	int32 msgId=1;
	int64 rid			= 2; //角色id
	int32 type			= 3; //类型 0登录，1重连 2跨服登录
	int32 gameType 	= 4; //游戏类型 101捕鱼达人
}

//登录子游戏
message LoginSubGameResponse{
	int32 msgId=1;
	int32 result		= 2; //结果
}

//退出子游戏
message QuitSubGameRequest{
	int32 msgId=1;	
	int64 rid 			= 2; //角色ID
}
//退出子游戏
message QuitSubGameResponse{
	int32 msgId=1;
	int32 result		= 2; //结果
}


//登录类型
enum LoginType{
    ACCOUNT             = 0; //账号登录
}
